sycl/include/sycl/kernel_bundle.hpp

```diff
@@ -57,10 +57,24 @@ class kernel_impl;
 
 template <typename KernelName> kernel_id get_kernel_id();
 
+// During device compilation mode the compiler does not yet know
+// what the kernels are named because that is exactly what its trying to
+// figure out during this phase. Therefore, we make the query declarations
+// unconditionally available regardless of Func in order to avoid errors about
+// missing functions. During host compilation which is after the #else clause
+// we can use SFINAE with the is_kernel trait to disable these functions when
+// instantiated with a Func that is not a kernel because at this point the
+// compiler knows what is a kernel and what is not.
+#ifdef __SYCL_DEVICE_ONLY__
+namespace ext::oneapi::experimental {
+template <auto *Func> kernel_id get_kernel_id();
+} // namespace ext::oneapi::experimental
+#else
```

Comment:
That is a lot of repeated words.

If we are not going with deferred diagnostics mechanism, can we then just change the definition of `is_kernel` trait to default to `true` for device compilation?

That way we will only need to repeat this comment once